binding/Binding/SKTypeface.cs

```diff
@@ -108,6 +110,54 @@ public int CharsToGlyphs (IntPtr str, int strlen, SKEncoding encoding, out ushor
 				}
 			}
 		}
+
+		public string FamilyName {
+			get {
+				var r = SkiaApi.sk_typeface_get_family_name(Handle);
+				try {
+					var cstr = SkiaApi.sk_string_get_c_str(r);
+					var clen = SkiaApi.sk_string_get_size(r);
+					return Util.GetString(cstr, (int)clen, SKTextEncoding.Utf8); 
+				} finally {
+					SkiaApi.sk_string_destructor(r);
+				}
+			}
+		}
+
+		public UInt32[] GetTableTags()
+		{
+			int tableCount = SkiaApi.sk_typeface_count_tables(Handle);
+			UInt32[] result = new UInt32[tableCount];
+			int r = SkiaApi.sk_typeface_get_table_tags(Handle, result);
+			if (r == 0) {
+				throw new Exception("Unable to read the tables for the file.");
```

Comment:
About the count_tables/get_table_tags duo, I tried to keep the C Api as close as possible to the C++ api, so there are 2 methods indeed. But the only reason I can think you would want to know the table count is so you can allocate memory and then call GetTableTags. It seems just the classing workaround for C arrays not having a length.

If you look at the comments in the C++ methods:
```
    /** Return the number of tables in the font. */
    int countTables() const;

    /** Copy into tags[] (allocated by the caller) the list of table tags in
     *  the font, and return the number. This will be the same as CountTables()
     *  or 0 if an error occured. If tags == NULL, this only returns the count
     *  (the same as calling countTables()).
     */
    int getTableTags(SkFontTableTag tags[]) const;

```
I think the idea is that you call CountTables, allocate an array of ints, and then call GetTableTags with the allocated array. You might even forget about CountTables and call twice GetTableTags: the first time to know the memory to allocate, the second to actually get the tags. I don't see any other reason why you would want to call CountTables alone.

Currently, if you want to know the tables in a font from C#, you would call:
```
  int[] tables = typeface.GetTableTags();
```

If we changed it to follow the C++ api, you would have to write:

```
int[] tables = new int[typeface.GetTableCount()];
typeface.GetTableTags(ref tables);
```

The first approach I think is more inline with what a C# user would expect, the second is more what a C user would expect. (and that is why I kept the C Api that way, similar to the C++ API).

A side note: SKIA internally uses FreeType and CoreGraphics to return the tables and tablecount. Coregraphics provides for getting the tables just returns an array too, there is no CountTables there:
https://developer.apple.com/library/mac/documentation/Carbon/Reference/CTFontRef/#//apple_ref/c/func/CTFontCopyAvailableTables

So if you look at SkTypeface_Mac::onGetTableTags(SkFontTableTag tags[]) you'll see it just gets an array with the tags, then returns the count as the array length.
In freetype there are indeed 2 calls to the same method ( FT_Sfnt_Table_Info ), but it is just using the standard C approach: First call with null to get the table count, then a second call to get the actual data.

About the exception, well, it can be indeed changed, but I don't think in this particular case it would be a problem: It should never fire. All fonts have a number of required tables, and a font without any table wouldn't have any data. I am not 100 % if there is any font format that doesn't have tables, but the common ones (ttf, otf, etc) have a number of required tables like the font header. A return of 0 or null here should be exceptional, and probably caused by a corrupt font or out of memory condition. I think raising an exception here makes sense so you don't have to check for null (or 0 tables) since that means a bigger error anyway, most likely one you can't handle.

What I am not 100% sure is about the exception in  GetTableData. I originally designed it so it owuld throw an Exception if the table tag doesn't exist, but this proved to be a problem even in my own code, where I regularly want to try to get data from optional tables and I don't want to first retrieve the list of available font tables with GetTableTags, and see if the tag is present in the array to only then call GetTableData. 

What I did was to add a new method TryGetTableData, which is similar to GetTableData but will return null instead of an exception. It follows the Parse/TryParse or GetValue/TryGetValue pattern in .NET, where you have a method that returns an exception, but if you need to call that method many times and it can fail, it is better to call TryMethod instead of Method and catch the exceptions. But we could modify GetTableData to return null if the tag doesn't exists (and so remove TryGetTableData)